### DISCUSSION TITLE
How to filter errors in result

### DISCUSSION THREAD

**Discussion by CH-EricLundberg** [graphql-java/graphql-java#2969] [Q&A]
Curious where I could gain access to the errors from a query/mutation result so I can filter them. The goal is to remove duplicates.  

Example running a query that has X fields that are all fetched with the same underlying rest api.  Don't want to have X error messages saying "Service down".   

I'm using spring-graphql for the wiring, but assume I can gain access to the graphql-java config in some fashion.

**Comment by bbakerman** [comment]
See `graphql.execution.instrumentation.Instrumentation#instrumentExecutionResult(graphql.ExecutionResult, graphql.execution.instrumentation.parameters.InstrumentationExecutionParameters, graphql.execution.instrumentation.InstrumentationState)` which allows you to change the result object before its returned

**Reply by CH-EricLundberg** [reply]
That looks great, any thoughts on how to do it in 18.3 (which seems to be what the spring-graphql 1.0.X is on?) I can 'getErrors' from the result but it is an immutable list

**Reply by CH-EricLundberg** [reply]
I took another look and my somewhat ganky approach is to get the immutable errors, copy them into a mutable list and do the filtering on that, then make a new CompletableFuture ala:

return CompletableFuture.completedFuture(new ExecutionResultImpl(executionResult.getData(), mutableErrors, executionResult.getExtensions()));

I suspect you could do a filter on the spring graphql side with a custom WebGraphQlInterceptor  public Mono<WebGraphQlResponse> intercept(WebGraphQlRequest request, WebGraphQlInterceptor.Chain chain) {
     } but that starts to feel a bit implementation dependent.
